### DISCUSSION TITLE
How to solve the warning: Attempted to read state from uninitialized service

### DISCUSSION THREAD

**Discussion by nikgraf** [statelyai/xstate#3632] [Q&A]
```
Warning: Attempted to read state from uninitialized service 'rootScreen'. Make sure the service is started first.
```

<img width="593" alt="Screenshot 2022-10-09 at 08 46 08" src="https://user-images.githubusercontent.com/223045/194742171-8b404ee5-dd8a-47b3-a5c6-4a269e247d66.png">

I nailed it down to the fact that it only happens when the final redirect is happening https://github.com/SerenityNotes/Serenity/blob/4a30e8bcc90afdc2eef3893770e0e660286506b5/apps/app/navigation/screens/rootScreen/rootScreenMachine.ts#L185-L195. The screen where useMachine is used https://github.com/SerenityNotes/Serenity/blob/4a30e8bcc90afdc2eef3893770e0e660286506b5/apps/app/navigation/screens/rootScreen/RootScreen.tsx#L9-L13 , then is not re-rending anymore and the state with this entry action is a final one.

I read about this https://github.com/statelyai/xstate/issues/1132#issuecomment-633718629, but also tried a stable object in the render function memoing the navigation.

Same is happening in this case https://github.com/SerenityNotes/Serenity/tree/4a30e8bcc90afdc2eef3893770e0e660286506b5/apps/app/navigation/screens/workspaceRootScreen. Seems like xstate is not handling the ReactNavigation redirect well 🙈

I tried to reproduce warning in isolation, but wasn't able to: https://codesandbox.io/s/xstate-reproduce-warning-zcbm79?file=/App.jsx

Any ideas what's causing it?

**Comment by Andarist** [comment]
It's hard to tell based on the given information. Does the problem occur in the web build? If yes then you could try to share https://www.replay.io/ recording with me so I could figure it out.
